server/api/blocks.go

```diff
@@ -782,3 +784,182 @@ func (a *API) handleDuplicateBlock(w http.ResponseWriter, r *http.Request) {
 }
+
+func (a *API) handleMoveBlockTo(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation POST /boards/{boardID}/blocks/{blockID}/move/{where}/{dstBlockID} moveBlockTo
+	//
+	// Move a block after another block in the parent card
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: boardID
+	//   in: path
+	//   description: Board ID
+	//   required: true
+	//   type: string
+	// - name: blockID
+	//   in: path
+	//   description: Block ID
+	//   required: true
+	//   type: string
+	// - name: where
+	//   in: path
+	//   description: Relative location respect destination block (after or before)
+	//   required: true
+	//   type: string
+	// - name: dstBlockID
+	//   in: path
+	//   description: Destination Block ID
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       type: array
+	//       items:
+	//         "$ref": "#/definitions/Block"
+	//   '404':
+	//     description: board or block not found
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	boardID := mux.Vars(r)["boardID"]
+	blockID := mux.Vars(r)["blockID"]
+	dstBlockID := mux.Vars(r)["dstBlockID"]
+	where := mux.Vars(r)["where"]
+	userID := getUserID(r)
+
+	if where != "after" && where != "before" {
+		message := fmt.Sprintf("invalid where parameter, use before or after")
+		a.errorResponse(w, r, model.NewErrBadRequest(message))
+	}
+
+	board, err := a.app.GetBoard(boardID)
+	if err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	if userID == "" {
+		a.errorResponse(w, r, model.NewErrUnauthorized("access denied to board"))
+		return
+	}
+
+	block, err := a.app.GetBlockByID(blockID)
+	if err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	if board.ID != block.BoardID {
+		message := fmt.Sprintf("block ID=%s on BoardID=%s", block.ID, boardID)
+		a.errorResponse(w, r, model.NewErrNotFound(message))
+		return
+	}
+
+	dstBlock, err := a.app.GetBlockByID(dstBlockID)
+	if err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	if board.ID != block.BoardID {
+		message := fmt.Sprintf("block ID=%s on BoardID=%s", block.ID, boardID)
+		a.errorResponse(w, r, model.NewErrNotFound(message))
+		return
+	}
+
+	if block.ParentID != dstBlock.ParentID {
+		message := fmt.Sprintf("invalid dstBlockId")
+		a.errorResponse(w, r, model.NewErrBadRequest(message))
+		return
+	}
+
+	if !a.permissions.HasPermissionToBoard(userID, boardID, model.PermissionManageBoardCards) {
+		a.errorResponse(w, r, model.NewErrPermission("access denied to modify board cards"))
+		return
+	}
+
+	card, err := a.app.GetBlockByID(block.ParentID)
+	if err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	contentOrder := card.Fields["contentOrder"].([]interface{})
+	newContentOrder := []interface{}{}
+	foundDst := false
+	foundSrc := false
+	for _, id := range contentOrder {
```

Comment:
I think this logic belongs in the app layer.  It can be deferred to a follow-up PR with a ticket.

Reply:
Yes, that's true, I think I would love to have an API about Content Blocks, instead of Blocks, and yes, that should have its counterpart in the App layer. Anyway, as I said, this is behind a feature flag and is for iterating over it whenever is finished, I don't want this to be the definitive API.

Anyway, I going to try to move it to the app layer now in a new `content_blocks.go` in the app, and one in the `api`, and add some other APIs.

Reply:
Done, PTAL to that part, I changed it to make it more open to grow later.

Reply:
Looks great.  Thanks @jespino.  I added some suggestions to fix linter.  Good for merge after that.